src/smc-util/misc.coffee

```diff
@@ -1898,3 +1898,17 @@ exports.bind_objects = (scope, arr_objects) ->
                 return bound_func
             else
                 return val
+
+# Provide the index that a value would be inserted
+# into an array where the val is between the number before
+# and after it
+exports.array_bisect = (arr, val) ->
```

Comment:
Takes an Array and a Val. Returns the first index where the value there is greater than `Val`. Else returns the length of the array.

Or you can say something about assumptions of sortedness, but the code does no such thing.